driver.go

```diff
@@ -15,16 +15,22 @@ type SnowflakeDriver struct{}
 // Open creates a new connection.
 func (d SnowflakeDriver) Open(dsn string) (driver.Conn, error) {
 	glog.V(2).Info("Open")
-	var err error
-	sc := &snowflakeConn{
-		SequenceCounter: 0,
-	}
 	ctx := context.TODO()
-	sc.cfg, err = ParseDSN(dsn)
+	cfg, err := ParseDSN(dsn)
 	if err != nil {
-		sc.cleanup()
 		return nil, err
 	}
+	return d.OpenWithConfig(ctx, *cfg)
+}
+
+// OpenWithConfig creates a new connection with the given config.
+func (d SnowflakeDriver) OpenWithConfig(ctx context.Context, config Config) (driver.Conn, error) {
```

Comment:
For this to be useful outside the driver you'll also need to implement the driver.Connector interface. See: https://golang.org/pkg/database/sql/#OpenDB

This is how we connect to Snowflake using a connector:
https://github.com/snowflakedb/snapps/blob/master/src/nulib/snowflake.go#L186

Also I think this'll need tests. Specifically around making sure that you can create identical connections using `sql.Open()` and `sql.OpenDB()`.

Reply:
Ah, I see, let me make it support that interface.

Reply:
JFYI we needed to add a Copy() method to the config for it to work properly. You may not need this but the thing to be careful of is if the config has any pointers or container types like maps or slices.

See:
https://github.com/snowflakedb/snapps/blob/master/src/nulib/gosnowflake/driver.go#L34
https://github.com/snowflakedb/snapps/blob/master/src/nulib/gosnowflake/dsn.go#L60

Reply:
Yep, I realized the `Config` struct has all public fields, some of which are mutable.
I feel like concepts like configuration are meant to be immutable, so perhaps its better to just make all fields private, and only let people use a builder to create instances of config. That way, external users are guaranteed not to mutate the config while the driver is running.